src/Symfony/Component/DependencyInjection/Dumper/YamlDumper.php

```diff
@@ -104,6 +104,11 @@ private function addService($id, $definition)
             $code .= sprintf("        synchronized: true\n");
         }
 
+        if ($definition->isDeprecated()) {
+            $code .= "        deprecated: true\n";
+            $code .= sprintf("        deprecation_message: %s\n", $definition->getDeprecationMessage('%service_id%'));
```

Comment:
You mean `getDeprecationMessage`? My reasoning is the following: we internally have a template, and this template is resolved to a message by replacing the placeholder, which is what this method does. Thus the names: template when %service_id% is used, message when resolved. Does it make sense?